dubbo.go

```diff
@@ -17,15 +17,26 @@
 
 package dubbo
 
+import (
+	"sync"
+)
+
 import (
 	"github.com/pkg/errors"
 )
 
 import (
 	"dubbo.apache.org/dubbo-go/v3/client"
+	"dubbo.apache.org/dubbo-go/v3/common"
 	"dubbo.apache.org/dubbo-go/v3/server"
 )
 
+var (
```

Comment:
Add locks there to protect ```consumerServices``` and ```providerServices``` separately since ```dubbo.SetConsumerService``` and ```dubbo.SetProviderService``` could be invoked concurrently.